x-pack/plugins/uptime/public/components/overview/monitor_list/columns/monitor_tags.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { EuiBadge, EuiBadgeGroup } from '@elastic/eui';
+import { MonitorSummary } from '../../../../../common/runtime_types/monitor';
+import { Ping } from '../../../../../common/runtime_types/ping';
+import { EXPAND_TAGS_LABEL } from './translations';
+
+interface Props {
+  ping?: Ping | null;
+  summary?: MonitorSummary;
+}
+
+const getTagsFromSummary = (summary: MonitorSummary) => {
+  let tags = new Set<string>();
+  summary.state.summaryPings.forEach((ping) => {
+    tags = new Set([...tags, ...(ping?.tags ?? [])]);
+  });
+
+  return [...tags];
+};
+
+const getTagsFromPing = (ping: Ping) => {
+  return ping?.tags ?? [];
+};
+
+export const MonitorTags = ({ ping, summary }: Props) => {
+  const [toDisplay, setToDisplay] = useState(5);
+  let tags: string[];
+
+  if (summary) {
```

Comment:
What are the situations where the summary doesn't exist?

Reply:
on monitor list we use summary